File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.swift
import Foundation

func solution(_ price:Int, _ money:Int, _ count:Int) -> Int64{
    var totalPrice: Int64 = 0
    
    for cnt in 1...count {
        totalPrice += Int64(price * cnt)
    }
    
    return totalPrice - Int64(money) <= 0 ? 0 : totalPrice - Int64(money)
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cpp
using namespace std;

long long solution(int price, int money, int count) {
    long long answer = 0, sum = 0;

    for (int i = 1; i <= count; i++) {
        sum += (price * i);
    }

    answer = money - sum;

    return (answer >= 0) ? 0 : -answer;
}